src/simplify_algebra.cpp

```diff
@@ -72,14 +74,15 @@ struct find_mul_conv
         auto a_ins    = r.instructions["a"];
         auto w_ins    = r.instructions["w"];
 
-        auto broadcast_op = any_cast<op::broadcast>(a_ins->get_operator());
-        if(broadcast_op.axis != 1)
+        auto a_strides = a_ins->get_shape().strides();
+        if(a_strides.at(0) != 0 or a_strides.at(1) != 1 or
+           std::any_of(
+               a_strides.cbegin() + 2, a_strides.cend(), [](std::size_t i) { return i != 0; }))
             return;
 
+        auto sq    = m.insert_instruction(ins, make_op("squeeze"), a_ins->inputs().front());
```

Comment:
squeeze would only remove axes with dims 1.  it is possible that axis has strides 0 but  dims != 1.

Reply:
Let me make sure I understand. The squeeze is applying on the input to previous `broadcast` or `multibroadcast` instruction. It's a `nop` for the `broadcast` case. For `multibroadcast` the `squeeze` should get rid of the single-dimensional entries. But there might be a problem when this gets matched for a case like this:
```
x = {256, 14, 14}, {1, 0, 0}
a = multibroadcast(x) => {4, 256, 14, 14}, {0, 1, 0, 0}
c = conv(x, w)
mul(c, a)
```
For this case the squeeze and broadcast won't work.

Reply:
I made some tests for this case, looks like the code on develop also fails for this case.

Reply:
> x = {256, 14, 14}, {1, 0, 0}
a = multibroadcast(x) => {4, 256, 14, 14}, {0, 1, 0, 0}
c = conv(x, w)
mul(c, a)

I was thinking about similar case. In this case, `x` itself is broadcasted, you can keep traversing back in graph (with some checks) to find real input. 

We should keep some check (`squeeze_out.shape.lens().size() ==1 ?` )at least to avoid applying matcher in such case rather than failing at broadcast.